### DISCUSSION TITLE
Is it possible to undo/revert a synchronization

### DISCUSSION THREAD

**Discussion by philipco** [abraunegg/onedrive#2124] [Q&A] [answered]
Hello,

I ran `onedrive --synchronize`, it overwrote my local files with remote files that were out-of-date. I don't know how to recover these files. 
Is there a way to undo/revert a synchronization ? 

Or do you have any ideas what can I do ? 

Thanks a lot !

**Comment by abraunegg** [comment, marked as answer]
@philipco 
> Or do you have any ideas what can I do ?

Firstly - what application version are you running and what is your configuration? Output of `onedrive --version` and `onedrive --display-config` would be helpful.

Secondly, the application *by default*, will not overwrite local files if a file of the same name exists locally - it will generate a conflict, and, create a *backup* of the local file that will have hostname of the system included in the file name.

My suggestion is to potentially look for these backup files - as this might be the data you are looking for.

**Reply by philipco** [reply]
Hello,

Thank you very much for your helpful answer.
Here are the outputs of the command: 
```
onedrive version                       = v2.4.2-11-g42bc0bf
Config path                            = /home/.config/onedrive
Config file found in config path       = false
Config option 'check_nosync'           = false
Config option 'sync_dir'               = /home/OneDrive
Config option 'skip_dir'               = 
Config option 'skip_file'              = ~*|.~*|*.tmp
Config option 'skip_dotfiles'          = false
Config option 'skip_symlinks'          = false
Config option 'monitor_interval'       = 300
Config option 'min_notify_changes'     = 5
Config option 'log_dir'                = /var/log/onedrive/
Config option 'classify_as_big_delete' = 1000
Config option 'sync_root_files'        = false
Selective sync configured              = true
```

Unfortunately, the folder /var/log/onedrive does not exist on my device. Reading, the USAGE.md document, I guess that it is because I have not enable loggings before? 
I just enabled it doing:
```
onedrive --enable-logging
sudo mkdir /var/log/onedrive
sudo chown root.users /var/log/onedrive
sudo chmod 0775 /var/log/onedrive
```
I suppose that in this case there is no backup? Is there any other option to recover my files?

Thank a lot,
Best regards

**Comment by abraunegg** [comment]
@philipco 
> v2.4.2-11-g42bc0bf

You are running a VERY old and unsupported client version. As per the [installation guide](https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md):

![image](https://user-images.githubusercontent.com/4956234/184225216-fd1f5842-2328-49d7-826e-0d9f5b97012e.png)

The client version you are running - you installed manually, sometime between May 2020 and June 2020 .. so your client version is over 2 years old ... I suggest that you *urgently* update your client software.

If you use a common distribution, install the client from the distribution package.

If you use Debian or Ubuntu - you must install following this process: https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md

> Unfortunately, the folder /var/log/onedrive does not exist on my device. Reading, the USAGE.md document, I guess that it is because I have not enable loggings before?

That is not how you enable logging fully ... please re-read the usage document.

> I suppose that in this case there is no backup? Is there any other option to recover my files?

If the files do not exist - but they should .. but given your old client version - and that you have not kept your client up-to-date - you potentially were hit by this bug:
* https://github.com/abraunegg/onedrive/issues/1714
* https://github.com/abraunegg/onedrive/pull/1717

This was fixed in client version v2.4.14 - https://github.com/abraunegg/onedrive/releases/tag/v2.4.14 - released November 2021
